src/servers/src/otlp/logs.rs

```diff
@@ -0,0 +1,543 @@
+        }
+    }
+}
+
+fn scope_to_pipeline_value(
+    scope: Option<InstrumentationScope>,
+) -> (PipelineValue, PipelineValue, PipelineValue) {
+    scope
+        .map(|x| {
+            (
+                PipelineValue::Map(Map {
+                    values: key_value_to_map(x.attributes),
+                }),
+                PipelineValue::String(x.version),
+                PipelineValue::String(x.name),
+            )
+        })
+        .unwrap_or((
+            PipelineValue::Null,
+            PipelineValue::Null,
+            PipelineValue::Null,
+        ))
+}
+
+fn scope_to_jsonb(
+    scope: Option<InstrumentationScope>,
+) -> (JsonbValue<'static>, Option<String>, Option<String>) {
+    scope
+        .map(|x| {
+            (
+                key_value_to_jsonb(x.attributes),
+                Some(x.version),
+                Some(x.name),
+            )
+        })
+        .unwrap_or((JsonbValue::Null, None, None))
+}
+
+fn log_to_pipeline_value(
+    log: LogRecord,
+    resource_schema_url: PipelineValue,
+    resource_attr: PipelineValue,
+    scope_schema_url: PipelineValue,
+    scope_name: PipelineValue,
+    scope_version: PipelineValue,
+    scope_attrs: PipelineValue,
+) -> PipelineValue {
+    let log_attrs = PipelineValue::Map(Map {
+        values: key_value_to_map(log.attributes),
+    });
+    let mut map = HashMap::new();
+    map.insert(
+        "Timestamp".to_string(),
+        PipelineValue::Uint64(log.time_unix_nano),
+    );
+    map.insert(
+        "ObservedTimestamp".to_string(),
+        PipelineValue::Uint64(log.observed_time_unix_nano),
+    );
+
+    // need to be convert to string
+    map.insert(
+        "TraceId".to_string(),
+        PipelineValue::String(bytes_to_hex_string(&log.trace_id)),
+    );
+    map.insert(
+        "SpanId".to_string(),
+        PipelineValue::String(bytes_to_hex_string(&log.span_id)),
+    );
+    map.insert("TraceFlags".to_string(), PipelineValue::Uint32(log.flags));
+    map.insert(
+        "SeverityText".to_string(),
+        PipelineValue::String(log.severity_text),
+    );
+    map.insert(
+        "SeverityNumber".to_string(),
+        PipelineValue::Int32(log.severity_number),
+    );
+    // need to be convert to string
+    map.insert(
+        "Body".to_string(),
+        log.body
+            .as_ref()
+            .map(|x| PipelineValue::String(log_body_to_string(x)))
+            .unwrap_or(PipelineValue::Null),
+    );
+    map.insert("ResourceSchemaUrl".to_string(), resource_schema_url);
+
+    map.insert("ResourceAttributes".to_string(), resource_attr);
+    map.insert("ScopeSchemaUrl".to_string(), scope_schema_url);
+    map.insert("ScopeName".to_string(), scope_name);
+    map.insert("ScopeVersion".to_string(), scope_version);
+    map.insert("ScopeAttributes".to_string(), scope_attrs);
+    map.insert("LogAttributes".to_string(), log_attrs);
+    PipelineValue::Map(Map { values: map })
+}
+
+fn build_identity_schema() -> Vec<ColumnSchema> {
```

Comment:
by identity pipeline, I prefer it's an auto generated pipeline and schema based on input data. Here we already manually defined a schema for otel logs. So we better not to name it and assume it the *identity* pipeline

Reply:
I've considered ways to automatically generate a pipeline. But it still has to convert the `ExportLogsServiceRequest` into a `PipelineValue` before it can be processed, so for performance reasons I skipped that step and just converted the ExportLogsServiceRequest into a Rows

Reply:
I would prefer using a different name other than `identity`, maybe something like `pre_defined` or `otlp_logs`, so the semantics of `identity` actually remains same within the project.